### ISSUE TITLE
Multi-file download feature not available

### ISSUE THREAD

**Issue by michaellegal** [filesender/filesender#998]
Hi, we are running FileSender 2.24 and the documentation says the multi-file download at once is available but I cannot find that feature.
May be it's important to mention that file encryption has been set up as mandatory.

Best regards,

Michael

**Comment by meijer** [comment]
Hi Michael, to make sure I understand what you're writing: you're running v2.24, you have encryption set up as mandatory, and your problem is that when you want to download a set of encrypted files, you can't download that entire set in one go?

**Comment by michaellegal** [comment]
Hi Jan, exactly.

Michael

[https://darma.crf.canon.fr/badge_editor/v2/icons/top.jpg]
Michael LE GAL
Acting ISS Manager and Chief Information Security Officer
Information System & Security Department
Support of Operations Division
CANON RESEARCH CENTRE FRANCE S.A.S.

Rue de la Touche Lambert - CS 91716
35517 Cesson-Sévigné Cedex - FRANCE
Tel.

:

+33 (0)2.99.87.68.55

Fax

:

+33 (0)2.99.84.11.30

E-mail

:

***@***.***


[https://darma.crf.canon.fr/badge_editor/v2/icons/bottom.jpg]
** Please consider the environmental impact before printing this e-mail. **

**Comment by monkeyiq** [comment]
Creating zip64 files from selected files of an encrypted upload may not be available in all browsers. It should show up in Chrome and Safari for example but some of the features it relies on were not available in Firefox at the time they were implemented. I plan to retest at some stage in Firefox as it would be very nice to be able to enable those features there as well.

As a first move, if it does not show up in a modern Chrome for you we can start trying to work out why.

**Comment by monkeyiq** [comment]
I don't think that encryption being mandatory will change things either way. I just set that and can download all files from an encrypted transfer as a zip in Chrome (using git development, though I don't think there have been fixes relevant to this issue from 2.24-dev). At times it can be hard to replicate the issue locally which makes it harder for me to try to find a remedy.

Looking at the code and with links to version 2.24 source the key block of code here is the isEncrypted/allowStreamSaver block of code 

https://github.com/filesender/filesender/blob/14b83014cd11c101234f37336b03890ed76254de/templates/download_page.php#L50
 
The allowStreamSaver is enabled by a single config option `streamsaver_enabled` (true by default, linked below) and then specifically for browsers in the code block at:

https://github.com/filesender/filesender/blob/14b83014cd11c101234f37336b03890ed76254de/classes/utils/Browser.class.php#L60

It would be very useful to debugging this if you could see the output of the following line executed in the Browser constructor linked above to make sure the browser is being detected properly there
```
Logger::error('is chrome ' . $this->isChrome );
```
This should put information in your /opt/filesender/logs directory depending on your logging configuraiton in config.php.

By default in 2.24 streamsaver should be enabled and these tests should allow it to run on most browsers I had consistent positive results for as shown in the default configuration shown at

https://github.com/filesender/filesender/blob/14b83014cd11c101234f37336b03890ed76254de/includes/ConfigDefaults.php#L274

You might also force the `$this->allowStreamSaver = true;` at the end of the Browser __construct method to see the buttons to download as a zip and see if things work from there. If so we can try to work out why allowStreamSaver is not being set correctly in some cases.

**Comment by monkeyiq** [comment]
I am on the localhost/filesender/?s=download page here. I should also mention that the download archive button does not appear for transfers with only one file (which I mention because I ran into it myself while testing a while back).

**Comment by monkeyiq** [comment]
I am wondering if this issue is still active?

**Comment by WebSpider** [comment]
This is related to multi-file download only being available for encrypted transfers when using streamsaver. 
I will make sure we have this documented.

**Comment by WebSpider** [comment]
This behaviour has since been documented. If you feel this is still an issue, feel free to reopen.
